docs/adding-a-library.md

```diff
@@ -0,0 +1,312 @@
+# Integrating a library within the hub
+
+The Hugging Face hub aims to facilitate the sharing of machine learning models, checkpoints, and artifacts. 
+This endeavor starts with the integration of its tool stack within downstream libraries, and we're happy to announce 
+the fruitful collaboration between Hugging Face and SpaCy, AllenNLP, Timm, among many other incredible libraries.
+
+> For a full list of supported libraries see [the following table](https://github.com/huggingface/huggingface_hub/blob/96e49483ace685cf526492b248ba3341bc1b1564/docs/libraries.md).
+
+We believe the model hub is a step in the correct direction for several reasons. It offers:
+
+- Free model hosting for libraries and their users. Custom plans exist for users wishing to keep their models 
+  private but keep the ease of use.
+- Built-in file versioning, even with very large files, thanks to a git-based approach
+- Hosted inference API for all models publicly available
+- In-browser widgets to play with the uploaded models.
+
+Thanks to these, we hope to achieve true shareability across the machine learning ecosystem, reproducibility, 
+and the ability to offer simple solutions directly from the browser. To that end, we're looking to make it very 
+simple to integrate the hub within downstream libraries or standalone machine learning models.
+
+The approach can be split in three different steps detailed below:
+
+- The "downstream" approach: downloading files from the hub so that they may be used simply locally.
+- The "upstream" approach: creating repositories and uploading files to the hub directly from your library.
+- Setting up the inference API for uploaded models.
+
+Before getting started, we recommend you first [create a HuggingFace account](https://huggingface.co/join). This will 
+create a namespace on which you can create repositories, and to which you can upload files to.
+
```

Comment:
Maybe we can add something along these lines:

If your library already has a set of existing models, you can create an organization where users would find the canonical models.

(No strong opinion though)